numba/targets/arrayobj.py

```diff
@@ -622,6 +623,93 @@ def array_argmax_impl(arry):
     return context.compile_internal(builder, array_argmax_impl, sig, args)
 
 
+def _np_round_intrinsic(tp):
+    # np.round() always rounds half to even
+    return "llvm.rint.f%d" % (tp.bitwidth,)
+
+def _np_round_float(context, builder, tp, val):
+    llty = context.get_value_type(tp)
+    module = cgutils.get_module(builder)
+    fnty = lc.Type.function(llty, [llty])
+    fn = module.get_or_insert_function(fnty, name=_np_round_intrinsic(tp))
+    return builder.call(fn, (val,))
+
+@builtin
+@implement(numpy.round, types.Kind(types.Float))
+def scalar_round_unary(context, builder, sig, args):
+    return _np_round_float(context, builder, sig.args[0], args[0])
+
+@builtin
+@implement(numpy.round, types.Kind(types.Integer))
+def scalar_round_unary(context, builder, sig, args):
+    return args[0]
+
+@builtin
+@implement(numpy.round, types.Kind(types.Complex))
+def scalar_round_unary_complex(context, builder, sig, args):
+    fltty = sig.args[0].underlying_float
+    cplx_cls = context.make_complex(sig.args[0])
+    z = cplx_cls(context, builder, args[0])
+    z.real = _np_round_float(context, builder, fltty, z.real)
+    z.imag = _np_round_float(context, builder, fltty, z.imag)
+    return z._getvalue()
+
+@builtin
+@implement(numpy.round, types.Kind(types.Float), types.Kind(types.Integer))
+@implement(numpy.round, types.Kind(types.Integer), types.Kind(types.Integer))
+def scalar_round_binary_float(context, builder, sig, args):
+    def round_ndigits(x, ndigits):
+        if math.isinf(x) or math.isnan(x):
+            return x
+
+        # NOTE: this is CPython's algorithm, but perhaps this is overkill
+        # when emulating Numpy's behaviour.
```

Comment:
It seems that Numpy doesn't worry about overflow, and just does `round(x * 10**ndigits) / 10**ndigits`: https://github.com/numpy/numpy/blob/master/numpy/core/src/multiarray/calculation.c#L593

In the implementation of atomic max for CUDA, we decided to use Numpy's semantics rather than CPython's semantics - should we do the same here for consistency?